### PR TITLE
Some CLI Fixes

### PR DESCRIPTION
**Pull request by CjBuresch**
+ some more help text
+fixed some problems with address input 
+ added a --check option so that we can unit test CLI input at some point

### CODE REVIEW

File: src/blackdoor/cqbe/cli/dh256.java [hunk 1]
```diff
@@ -126,13 +175,21 @@ public static void join(String[] args) {
...
+						node.setBootstrapNode(new L3Address(InetAddress
+								.getByName(bootstrapSplit[0]), Integer
+								.parseInt(bootstrapSplit[1])));
+					} else
+						throw new InvalidFormatException("Bootstrap ' "
+								+ bootstrap + " ' is formatted incorrectly. ");
+
 				}
-				node.buildNode();
+				if (!out.containsKey("check"))
```
Comment (kag0) [inline src/blackdoor/cqbe/cli/dh256.java]: what's this?

File: src/blackdoor/cqbe/cli/dh256.java [hunk 2]
```diff
@@ -172,24 +234,29 @@ public static void insert(String[] args) {
...
+								InetAddress.getByName(bootstrapSplit[0]),
+								Integer.parseInt(bootstrapSplit[1]));
+						router = Router.fromBootstrapNode(a);
+					} else {
+						router = Router.fromDefaultLocalNode();
+					}
+					int numberStored = router.put(fileAddress, fileBytes);
+					if (numberStored > 0)
+						System.out.println("file added successfully by "
+								+ numberStored + " nodes with address: "
```
Comment (kag0) [inline src/blackdoor/cqbe/cli/dh256.java]: awk/confusing phrasing
"file with address <> successfasdfasly stored by x nodes" might be better

File: src/blackdoor/cqbe/cli/dh256.java [hunk 3]
```diff
@@ -228,36 +294,59 @@ public static void retrieve(String[] args) {
...
-					L3Address a = new L3Address(
-							InetAddress.getByName(bootstrapSplit[0]),
-							Integer.parseInt(bootstrapSplit[1]));
-					router = Router.fromBootstrapNode(a);
+
+					String bootstrap = parsedArgs.get("bootstrap").getValues()
+							.get(0);
+					if (bootstrap.contains(":")) {
+						String[] bootstrapSplit = bootstrap.split(":");
+						L3Address a = new L3Address(
```
Comment (kag0) [inline src/blackdoor/cqbe/cli/dh256.java]: we do this so many times. DRY?

**Comment by CjBuresch** [comment]
made the changes you pointed out. Thoughts?

**Comment by euphzji** [comment]
Wayyyy prettier. I like.

**Pull request merged**